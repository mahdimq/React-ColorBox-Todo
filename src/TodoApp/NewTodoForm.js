import React, { useState } from 'react';

function NewTodoForm({ addTodo }) {
	const [formData, setFormData] = useState({ todo: '' });

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((data) => ({
			...data,
			[name]: value
		}));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		addTodo(formData);
		setFormData({ todo: '' });
	};

	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor='todo'>Enter Todo</label>
			<input
				id='todo'
				type='text'
				name='todo'
				placeholder='Wash car...'
				value={formData.todo}
				onChange={handleChange}
			/>

			<button>Add todo</button>
		</form>
	);
}

export default NewTodoForm;
