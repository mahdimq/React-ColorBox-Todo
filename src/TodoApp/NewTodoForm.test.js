import React from 'react';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// SMOKE TEST
it('renders without crashing', function () {
	render(<NewTodoForm />);
});

// SNAPSHOT TEST
it('matches snapshot', function () {
	const { asFragment } = render(<NewTodoForm />);
	expect(asFragment()).toMatchSnapshot();
});
