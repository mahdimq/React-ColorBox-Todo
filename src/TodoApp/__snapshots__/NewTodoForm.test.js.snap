// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Enter Todo
    </label>
    <input
      id="todo"
      name="todo"
      placeholder="Wash car..."
      type="text"
      value=""
    />
    <button>
      Add todo
    </button>
  </form>
</DocumentFragment>
`;
